sympy/printing/fcode.py

```diff
@@ -129,6 +131,33 @@ def _lead(self):
         else:
             raise ValueError("Unknown source format: %s" % self._settings['source_format'])
 
+    def _print_Symbol(self, expr):
+        if self._settings['name_mangling'] is True:
+            from sympy import symbols, ordered, Symbol
+            for sym in tuple(ordered(expr.atoms(Symbol))):
+                if sym not in self.mangled_symbols:
+                    name = sym.name
+                    while name.lower() in self.used_name:
+                        name += '_'
+                    self.used_name.append(name.lower())
+                    if name == sym.name:
+                        self.mangled_symbols[sym] = sym
+                    else:
+                        self.mangled_symbols[sym] = symbols(name, cls=sym.__class__, **sym.assumptions0)
```

Comment:
just wondering: do we need `symbols` here? (or can we simply use `Symbol`?)

Reply:
I tried `Symbol` here but it is not accepting these arguments.
It gives `TypeError: __new__() got multiple values for argument 'cls'`

Reply:
The reason being, if sym is of class Idx (for example), then the replacement should also be Idx. The function `symbols` is able to create such things since it takes cls argument.

Reply:
Oh yes, forgot about that. We should probably add a test for that specific case so that it won't be overlooked in e.g. a future refactoring.

Reply:
@bjodah  [this](https://github.com/jashan498/sympy/blob/1562b0e824129f6f00a500ab954f2febe7d2f83d/sympy/printing/tests/test_fcode.py#L187) assures that. It gives assertion error without it but still i am adding one more for that.